app/assets/javascripts/discourse/app/routes/application.js

```diff
@@ -259,7 +259,7 @@ const ApplicationRoute = DiscourseRoute.extend(OpenComposer, {
       window.location = getURL("/session/sso?return_path=" + returnPath);
     } else {
       this._autoLogin("login", {
-        notAuto: () => this.controllerFor("login").resetForm(),
+        notAuto: () => getOwner(this).lookup("controller:login").resetForm(),
```

Comment:
This is related to the mentioned `login` route hack, but is also a correction of (mis)usage of `controllerFor`. From the docs:

> Returns the controller of the current route, or a parent (or any ancestor) route in a route hierarchy.

In many cases where we're using this method the controller is not an ancestor 🥲 Also, it doesn't mean "give me the controller named X" but "give me the controller of a route named X" which isn't necessarily the same.